java/src/main/java/ai/rapids/cudf/DType.java

```diff
@@ -116,88 +268,111 @@ public boolean isBackedByInt() {
    *       DType.TIMESTAMP_NANOSECONDS
    */
   public boolean isBackedByLong() {
-    return LONGS.contains(this);
+    return LONGS.contains(this.typeId);
   }
 
+  /**
+   * Returns true if this type is backed by short type
+   * Namely this method will return true for the following types
+   *       DType.INT16,
+   *       DType.UINT16
+   */
+  public boolean isBackedByShort() { return SHORTS.contains(this.typeId); }
+
+  /**
+   * Returns true if this type is backed by byte type
+   * Namely this method will return true for the following types
+   *       DType.INT8,
+   *       DType.UINT8,
+   *       DType.BOOL8
+   */
+  public boolean isBackedByByte() { return BYTES.contains(this.typeId); }
+
+  /**
+   * Returns true if this type is backed by decimal type
+   * Namely this method will return true for the following types
+   *       DType.DECIMAL32,
+   *       DType.DECIMAL64
+   */
+  public boolean isBackedByDecimal() { return DECIMALS.contains(this.typeId); }
+
   /**
    * Returns true for duration types
    */
   public boolean isDurationType() {
-    return DURATION_TYPE.contains(this);
+    return DURATION_TYPE.contains(this.typeId);
   }
 
   /**
    * Returns true for nested types
    */
   public boolean isNestedType() {
-    return NESTED_TYPE.contains(this);
+    return NESTED_TYPE.contains(this.typeId);
   }
 
-  public int getNativeId() {
-    return nativeId;
+  public boolean isTimestamp() {
```

Comment:
nit: can we update this to be `isTimestampType` so it is consistent with the other APIs?

Reply:
Note that `isTimestamp` is a pre-existing `DType` method.  Maybe add `isTimestampType` and deprecate this method?

Reply:
@jlowe thanks I missed that. I saw the comments other places asking for us to use isTimestamp and I assumed it was new.  Also this is a nit and if we keep it as it I am fine with it.

Reply:
Added `isTimestampType` and deprecated `isTimestamp`